src/SonarScanner.MSBuild.Common/AnalysisProperties/ListSettingsProvider.cs

```diff
@@ -30,27 +30,29 @@ public class ListPropertiesProvider : IAnalysisPropertyProvider
 {
     private readonly IList<Property> properties;
 
+    public PropertyProviderKind ProviderType => PropertyProviderKind.SQ_SERVER_SETTINGS;
+
     #region Public methods
 
     public ListPropertiesProvider()
```

Comment:
This provider takes the properties from somewhere. I would make `PropertyProviderKind` a mandatory parameter, so the caller decides what the origin is:
```suggestion
    public ListPropertiesProvider(PropertyProviderKind providerType)
```

Reply:
I set it with a default value otherwise it's a mess - and I make a mandatory parameter for lots of usages that is not really needed now.